.github/workflows/test.yml

```
@@ -0,0 +1,82 @@
+name: Run tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [ "3.10", "3.12" ]
+
+    services:
+      redis:
+        image: redis:latest
```

```suggestion
        image: redis:latest
        ports:
          - 6379/tcp
```
Sorry, I just made the same mistake you did before :))

---

It's not default? :/

---

ok, so that didn't work either. Reverting for now. Will revisit tomorrow.

---

Huh, it's even more idiotic than that:
```
  /usr/bin/docker create --name 360ad4d52fa644fa81e68fe274b67c52_redislatest_bb60f6 --label d92611 --network github_network_1300f662df0d4c599a633329f57ebbdf --network-alias redis -p 6379/tcp  -e GITHUB_ACTIONS=true -e CI=true redis:latest
  b820475136acec8a3f083aa24642970ff97c0e6751f186f9e24c729e8951b431
  /usr/bin/docker start b820475136acec8a3f083aa24642970ff97c0e6751f186f9e24c729e8951b431
  b820475136acec8a3f083aa24642970ff97c0e6751f186f9e24c729e8951b431
  /usr/bin/docker ps --all --filter id=b820475136acec8a3f083aa24642970ff97c0e6751f186f9e24c729e8951b431 --filter status=running --no-trunc --format "{{.ID}} {{.Status}}"
  b820475136acec8a3f083aa24642970ff97c0e6751f186f9e24c729e8951b431 Up Less than a second
  /usr/bin/docker port b820475136acec8a3f083aa24642970ff97c0e6751f186f9e24c729e8951b431
  6379/tcp -> 0.0.0.0:32768
  6379/tcp -> [::]:32768
```
How are you even supposed to deduce the randomly assigned port... Anyway this should work
```yaml
        image: redis:latest
        ports:
          - 6379:6379
```

---

Let's see if we can put it in the pod yaml anyway

---

pod yaml was ugly. Scrapping that idea and trying the `service`